ddl/ddl_api.go

```diff
@@ -3587,6 +3583,10 @@ func (d *ddl) dropIndex(ctx sessionctx.Context, ti ast.Ident, isPK bool, indexNa
 		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(ti.Schema, ti.Name))
 	}
 
+	var isPK bool
+	if indexName.L == strings.ToLower(mysql.PrimaryKeyName) {
```

Comment:
We can create a normal index with name `primary`:
```
create table tt(`primary` int);
alter table tt add index (`primary`);
```

Reply:
This seems to be the problem that the index name we used to create the index for the 
`primary` column is not compatible with MySQL
```
mysql> create table tt(`primary` int);
Query OK, 0 rows affected (0.02 sec)
mysql> alter table tt add index (`primary`);
Query OK, 0 rows affected (0.03 sec)
Records: 0  Duplicates: 0  Warnings: 0
mysql> show create table tt;
+-------+--------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                             |
+-------+--------------------------------------------------------------------------------------------------------------------------+
| tt    | CREATE TABLE `tt` (
  `primary` int(11) DEFAULT NULL,
  KEY `primary_2` (`primary`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+-------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

```
tidb>  create table tt(`primary` int);
Query OK, 0 rows affected (0.01 sec)
tidb> alter table tt add index (`primary`);
Query OK, 0 rows affected (0.04 sec)
tidb> show create table tt;
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                  |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------+
| tt    | CREATE TABLE `tt` (
  `primary` int(11) DEFAULT NULL,
  KEY `primary` (`primary`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```